### ISSUE TITLE
Pravega Standalone - Pravega Readers fails to read from Pravega streams

### ISSUE THREAD

**Issue by ShwethaSNayak** [pravega/pravega#6629]
**Problem description**
Follow https://cncf.pravega.io/docs/nightly/beginner_dev_guide/ to create "**Pravega - Creating Your First Application**". 
Writing events to Pravega stream is successful but while reading events from same stream fails with **java.lang.IllegalArgumentException: Illegal stream name**

**While creating readerGroup issue is observed**.

These sample reader program was working fine sometime back.

**Problem location**
While creating Reader Group,  "Create stream failed due to invalid stream name _RGreaderGroups" error in Pravega standalone logs 

>  INFO  i.p.c.s.r.g.v.ControllerServiceImpl - [requestId=-6198967599560299798] createStream called for stream examples/_RGreaderGroups.

**Error logs**

     

> Exception in thread "main" java.lang.RuntimeException: java.lang.IllegalArgumentException: Illegal stream name: StreamConfiguration(scalingPolicy=ScalingPolicy(scaleType=FIXED_NUM_SEGMENTS, targetRate=0, scaleFactor=0, minNumSegments=1), retentionPolicy=null, timestampAggregationTimeout=0)
  at io.pravega.common.concurrent.Futures.getAndHandleExceptions(Futures.java:206)
 at io.pravega.client.admin.impl.ReaderGroupManagerImpl.createStreamHelper(ReaderGroupManagerImpl.java:80)
 at io.pravega.client.admin.impl.ReaderGroupManagerImpl.createReaderGroup(ReaderGroupManagerImpl.java:92)
at devguide.PravegaStreamReader.main(PravegaStreamReader.java:28)

**Comment by shshashwat** [comment]
The root cause of the problem is:-
While using the Pravega standalone, if you're using the latest master which has additional changes and the client library is 0.9.0, there might be several changes and few of them might not go as expected. In this particular case, while creating a `readergroup` the client is calling a controller method `createStream` for creating internal stream `_RGreadergroupStream`, however the internal streams can be created only using `createStreamInternal` because the validations for the stream name are different in these two methods. The changes are not there in Pravega v0.9.0. And hence on client we see the error `java.lang.IllegalArgumentException: Illegal stream name`. 
So, basically there is no code change required but additionally we can add this note to the `beginner_dev_guide` to use the same version of Pravega in client and standalone.
@pbelgundi, please share your views
